modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/northbound/OpcUaToJsonConverter.java

```diff
@@ -67,30 +67,40 @@ public class OpcUaToJsonConverter {
     public static @NotNull ByteBuffer convertPayload(
             final @NotNull EncodingContext serializationContext,
             final @NotNull DataValue dataValue) {
+        return convertPayload(serializationContext, dataValue, false);
+    }
+
+    public static @NotNull ByteBuffer convertPayload(
+            final @NotNull EncodingContext serializationContext,
+            final @NotNull DataValue dataValue,
+            final boolean includeMetadata) {
         final Object value = dataValue.getValue().getValue();
         if (value == null) {
             return ByteBuffer.wrap(EMPTY_BYTES);
         }
         final JsonObject jsonObject = new JsonObject();
-        if (value instanceof final DataValue v) {
-            if (v.getStatusCode().getValue() > 0) {
-                jsonObject.add("statusCode", convertStatusCode(v.getStatusCode()));
+
+        // Extract metadata from the outer DataValue when includeMetadata is enabled
+        if (includeMetadata) {
+            if (dataValue.getStatusCode() != null && dataValue.getStatusCode().getValue() > 0) {
```

Comment:
includes statusCode only when value > 0 (`good` status - value = 0) is never included. 
intentional?  will users want to see successful status as well?